File: app/src/main/java/kr/co/jayhy/weather/realm/dao/WeatherDao.java
package kr.co.jayhy.weather.realm.dao;

import android.support.annotation.NonNull;

import io.realm.Realm;
import io.realm.RealmObject;
import io.realm.RealmQuery;
import io.realm.RealmResults;
import kr.co.jayhy.weather.item.WeatherItem;
import kr.co.jayhy.weather.realm.object.WeatherObject;

/**
 * Created by jhkim on 2017-08-14.
 */

public class WeatherDao {

    private Realm mRealm;

    public WeatherDao(Realm realm) {
        this.mRealm = realm;
    }

    public void current_save(final WeatherObject item) {
        mRealm.executeTransaction(new Realm.Transaction() {
            @Override
            public void execute(Realm realm) {
                WeatherObject object = realm.createObject(WeatherObject.class);
                object.setMain(item.getMain());
                object.setIcon_url(item.getIcon_url());
                object.setTemp(item.getTemp());
                object.setTemp_min(item.getTemp_min());
                object.setTemp_max(item.getTemp_max());
                object.setDt(item.getDt());
            }
        });
    }

    public RealmResults<WeatherObject> current_loadAll() {
        return mRealm.where(WeatherObject.class).findAll();
    }

    public RealmResults<WeatherObject> current_loadAllAsync() {
        return mRealm.where(WeatherObject.class).findAll();
    }

    public void current_remove(@NonNull final RealmObject object) {
        final RealmResults<WeatherObject> results = mRealm.where(WeatherObject.class).findAll();
        mRealm.executeTransaction(new Realm.Transaction() {
            @Override
            public void execute(Realm realm) {
                results.deleteAllFromRealm();
            }
        });
    }

    public void removeAll() {
        mRealm.executeTransaction(new Realm.Transaction() {
            @Override
            public void execute(Realm realm) {
            }
        });
    }


    public long current_count() {
        return mRealm.where(WeatherObject.class).count();
    }
}

File: kotlin/src/main/java/kr/co/jayhy/weather/realm/dao/WeatherDao.kt
package kr.co.jayhy.weather.realm.dao

import io.realm.Realm
import io.realm.RealmObject
import io.realm.RealmResults
import kr.co.jayhy.weather.realm.`object`.WeatherObject

/**
 * Created by jhkim on 2017-08-14.
 */

class WeatherDao(private val mRealm: Realm) {

    fun current_save(item: WeatherObject) {
        mRealm.executeTransaction { realm ->
            val `object` = realm.createObject(WeatherObject::class.java)
            `object`.main = item.main
            `object`.icon_url = item.icon_url
            `object`.temp = item.temp
            `object`.temp_min = item.temp_min
            `object`.temp_max = item.temp_max
            `object`.dt = item.dt
        }
    }

    fun current_loadAll(): RealmResults<WeatherObject> {
        return mRealm.where(WeatherObject::class.java).findAll()
    }

    fun current_loadAllAsync(): RealmResults<WeatherObject> {
        return mRealm.where(WeatherObject::class.java).findAll()
    }

    fun current_remove(`object`: RealmObject) {
        val results = mRealm.where(WeatherObject::class.java).findAll()
        mRealm.executeTransaction { results.deleteAllFromRealm() }
    }

    fun removeAll() {
        mRealm.executeTransaction { }
    }


    fun current_count(): Long {
        return mRealm.where(WeatherObject::class.java).count()
    }
}
